datafusion/physical-plan/src/sorts/sort.rs

```diff
@@ -874,53 +892,67 @@ impl ExecutionPlan for SortExec {
 
         trace!("End SortExec's input.execute for partition: {}", partition);
 
-        if let Some(fetch) = self.fetch.as_ref() {
-            let mut topk = TopK::try_new(
-                partition,
-                input.schema(),
-                self.expr.clone(),
-                *fetch,
-                context.session_config().batch_size(),
-                context.runtime_env(),
-                &self.metrics_set,
-                partition,
-            )?;
-
-            Ok(Box::pin(RecordBatchStreamAdapter::new(
-                self.schema(),
-                futures::stream::once(async move {
-                    while let Some(batch) = input.next().await {
-                        let batch = batch?;
-                        topk.insert_batch(batch)?;
-                    }
-                    topk.emit()
-                })
-                .try_flatten(),
-            )))
-        } else {
-            let mut sorter = ExternalSorter::new(
-                partition,
-                input.schema(),
-                self.expr.clone(),
-                context.session_config().batch_size(),
-                self.fetch,
-                execution_options.sort_spill_reservation_bytes,
-                execution_options.sort_in_place_threshold_bytes,
-                &self.metrics_set,
-                context.runtime_env(),
+        let sort_satisfied = self
```

Comment:
This is the same calculation as `self.execution_mode()`, right? Maybe we could call `self.execution_mode` here instead to be more efficient and ensure the calculations remained in sync

Reply:
It is similar but not exactly the same (i.e. execution mode is derived from `sort_satisfied` but AFAIK the reverse is not possible). I think @berkaysynnada tried this but it didn't work

Reply:
I have tried that but didn't work. Knowing the sort is bounded or unbounded does not mean sort is satisfied.